operators/multiclusterobservability/controllers/multiclusterobservability/observatorium.go

```diff
@@ -96,6 +101,14 @@ func GenerateObservatoriumCR(
 	newSpec := observatoriumCR.Spec
 	oldSpecBytes, _ := yaml.Marshal(oldSpec)
 	newSpecBytes, _ := yaml.Marshal(newSpec)
+
+	log.Info("handling storage resize for observatorium", "isResizeForbidden", isResizeForbidden)
+	// isResizeForbidden will reset after operator restart, double check with scAllowVolumeExpansion
+	if isResizeForbidden || !scAllowVolumeExpansion {
+		rollbackStorageResizeForObservatorium(&oldSpec, &newSpec)
```

Comment:
Need to apply the same logic for alertmanager.

Another point: How about just rollback the changes for mco cr directly if the storageclass cannot support resize feature? thoughts?

Reply:
In my opinion, we should not revert the MCO CR without notifying end-user.
But the currently solution is not the perfect, the user can only check the MCO CR logs for the reason why the PVC is not resized.

The best solution should be validating resize request with validation webhook, so that it can return the resize error and reason directly to the end user when they apply the MCO CR.

Reply:
Besides, it will be complicated if we use the revision to rollback the MCO CR, we also need to consider the other changes, for example, retention change, together with the storage size change.

Reply:
Yes. I agree that we should not revert any user's changes without notifying. What is effort to use validation webhook? that is right way I think. I remembered we already exposed webhook api in 2.3.

Reply:
> Besides, it will be complicated if we use the revision to rollback the MCO CR, we also need to consider the other changes, for example, retention change, together with the storage size change.
> 
> For alertmanager, the statefulset rollback logic will be much more complicated than the observatorium CR, I remove that part because it will cause other issues(infinite reconcile loop, storage size mess up)

Yes. I have the same concern about infinite reconcile loop.

Reply:
We can add validation webhook for 2.4, what about 2.3?

Reply:
How about keep it as a known issue in 2.3?